crates/hotshot-state-prover/src/circuit.rs

```diff
@@ -304,9 +391,17 @@ mod tests {
             fee_ledger_comm: F::default(),
             stake_table_comm: (F::default(), F::default(), F::default()),
         };
+        let bad_state_msg = [F::default(); 7];
+        let sig_for_wrong_state = schnorr_keys
+            .iter()
+            .map(|(key, _)| {
+                SchnorrSignatureScheme::<Config>::sign(&(), key, bad_state_msg, &mut prng)
+            })
+            .collect::<Result<Vec<_>, PrimitivesError>>()
+            .unwrap();
         let (bad_circuit, public_inputs) = StateUpdateBuilder::<F>::build(
             &st,
-            &[],
+            &sig_for_wrong_state,
             &bad_lightclient_state,
```

Comment:
Why do we need to use `bad_lightclient_state` if we already set `bad_state_msg` to be signed?